lib/sessionParticipants/index.js

```diff
@@ -20,16 +24,18 @@ function matchingIndex(issuer, sessionIndex, nameId){
  * used to find the correct Session Participant object which
  * represents the issuer of the previous mentions request/response.
  * 
- * @issuer {string}   The string as it was received in the SAML request/response
- * @sessionIndex {string}   The string as it was received in the SAML request/response. Only available in LogoutRequests
- * @cb     {function} The callback that will be called with '(err, sessionParticipant)'
+ * @issuer {string} The string as it was received in the SAML request/response
+ * @sessionIndices {string|string[]} An single string or array of strings representing the
```

Comment:
Any reason to not go with only `string[]`? As it stands this is a breaking change, we may as well avoid that duality.

Reply:
I was going back and forth about this for some time. There are valid cases where we might want to get a session participant by a single, exact match session index; for instance, when we load the state returned from the SP: https://github.com/auth0/node-samlp/blob/d4e01c8a10410483251fabf2544bc234e2b87bbe/lib/logout.js#L226

Of course, we can update the call to wrap the single session index in an array:

```js
options.sessionParticipants.get(state.issuer, [state.sessionIndex], state.nameId, ...)
```

I'm on the fence between both options but don't have a strong opinion about either - happy to go with either of them.

Given the use case I mentioned above, would you still be in favour of always calling `#get` with an array of session indices?

Reply:
> Given the use case I mentioned above, would you still be in favour of always calling #get with an array of session indices?

Yes because one is strictly more expressive than the other and Javascript runtime "overloading" is error-prone and you are requiring the implementors to implement it over and over again (the `Array.isArray` check specifically). You may as well require only the more generic implementation.

Another approach that you might consider is to add the new SPI as an optional method. The way that it works is that the caller has a default implementation when the optional method is not implemented. In this case node-samlp would fall back to multiple calls to `SessionParticipants.get`.
```
SessionParticipants.get(issuer: string, sessionIndex: string, nameId: string): Participant
SessionParticipants.getAll(issuer: string, sessionIndices: string[], nameId: string): Participant[] // optional
```

This strategy is commonly used to evolve APIs while retaining backward compatibility. Admittedly, there's not a big compatibility concern in this case but I thought I would mention this anyway as a useful technique for evolving SPIs.